lib/fizzy/parser.cpp

```diff
@@ -139,12 +141,18 @@ inline parser_result<Memory> parse(const uint8_t* pos, const uint8_t* end)
 
 parser_result<std::string> parse_string(const uint8_t* pos, const uint8_t* end)
 {
-    std::vector<uint8_t> value;
-    std::tie(value, pos) = parse_vec<uint8_t>(pos, end);
+    // NOTE: this is an optimised version of parse_vec<uint8_t>
+    uint32_t size;
+    std::tie(size, pos) = leb128u_decode<uint32_t>(pos, end);
+
+    if ((pos + size) > end)
```

Comment:
This, I would say, is going a bit too far with parentheses :laughing:

Reply:
How? It is only one pair.